### PR TITLE
Bugfix/vpf detection rework

### PR DESCRIPTION
**Pull request by wvanhaevre**
- **Changed the VPF detection** to a waiting event + timer setup instead of waiting event + pause event. It was observed that for some streams (mostly VOD) AVplayer does not trigger the pause (and even keeps incrementing the currentTime whereas the stream has stopped).

- Last VPF reset timestamp (marking the stream is fine) is kept as start of the longer interval for which errorLog is monitored. This replaces the fixed interval that sometimes (when player has sufficient buffer) fell out of the timeranges containing the actual errors.

- As requested by Conviva: **Prevent multiple fatal errors** being reported during 1 viewer session. as this would cause multiple sessions being created on the Conviva backend, with huge negative impact on the player SPI.

### CODE REVIEW

**Review by therama — APPROVED** [review]

File: Code/Conviva/Source/ConvivaConnector.swift [hunk 1]
```diff
@@ -22,7 +22,7 @@ public struct ConvivaConnector {
     
     init(conviva: ConvivaEndpoints, player: THEOplayer, externalEventDispatcher: THEOplayerSDK.EventDispatcherProtocol? = nil) {
         self.endPoints = conviva
-        
+        self.convivaVPFDetector.player = player
         if let endPoints = self.endPoints {
```
Comment (Dev1an) [inline Code/Conviva/Source/ConvivaConnector.swift:26]: Why is this needed? `self.endPoints` cannot be nil here because it always equals to conviva, right?

**Review by Dev1an — COMMENTED** [review]
Looks good to me.
I don't know why the `endPoints` property etc. in the `ConvivaConnector` should be optional but this is unrelated to the VPF work.

File: Code/Conviva/Source/ConvivaConnector.swift (hunk 1 above)
Comment (wvanhaevre) [reply Code/Conviva/Source/ConvivaConnector.swift]: I updated the code
